### PR TITLE
Prepare for Core merge

### LINKED ISSUE #32: Prepare for Core merge
### Is your enhancement related to a problem? Please describe.

In preparation for opening a PR for WP Core, there are few maintenance tasks that we'll need to complete. This issue should be used as a tracking ticket for collecting all of those tasks in one place.

## Acceptance Criteria

- Remove the workflow rules that allow the workflow to be run on `feature/**` branches.
- Log performance results to the production codehealth API instead of the test dashboard.
- Update the workflow to only record and publish the performance results on pushes to trunk (not during PRs)

**Note:** Before we can test this in core, we will also need to coordinate getting the project token for the dashboard set up in the WP Core repo.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by mukeshpanchal27**
This PR prepares the Automated performance testing branch ready for core merge.

The remaining part **Set up the project token for the dashboard in the WP Core repo** that we have to do in the [WordPress/wordpress-develop](https://github.com/WordPress/wordpress-develop/) repository.

Closes https://github.com/10up/wordpress-develop/issues/32.

### CODE REVIEW

File: .github/workflows/performance.yml
Comment (joemcgill) [inline .github/workflows/performance.yml]: These need to stay, just remove the feature branch part.

File: .github/workflows/performance.yml
Comment (joemcgill) [inline .github/workflows/performance.yml]: I think we want this to run on PRs too, but we'll only want to post to the dashboard if it's a push to event (not on pull requests).

```suggestion
    if: ${{ github.repository == 'WordPress/wordpress-develop' || github.event_name == 'pull_request' }}
```

File: .github/workflows/performance.yml
Comment (joemcgill) [inline .github/workflows/performance.yml:157]: We'll need to add an `if` here so this only runs on pushes to WordPress branches, not on PRs.

**Review by joemcgill — CHANGES_REQUESTED** [review]
Still need to handle a few changes here, but a good start.

File: .github/workflows/performance.yml
Comment (mukeshpanchal27) [reply .github/workflows/performance.yml]: Thanks @joemcgill, The changes reverted in https://github.com/10up/wordpress-develop/pull/41/commits/e66687a08628d2348048dba0ec6cddc19a6d2937

File: .github/workflows/performance.yml
Comment (mukeshpanchal27) [reply .github/workflows/performance.yml]: Thanks Joe for review. Updated in https://github.com/10up/wordpress-develop/pull/41/commits/e66687a08628d2348048dba0ec6cddc19a6d2937

File: .github/workflows/performance.yml
Comment (mukeshpanchal27) [reply .github/workflows/performance.yml:157]: Added `if` condition in https://github.com/10up/wordpress-develop/pull/41/commits/e66687a08628d2348048dba0ec6cddc19a6d2937

**Comment by mukeshpanchal27** [comment]
@joemcgill Thank you for the review feedback. I have addressed all the feedback and PR is ready for review.

**Review by joemcgill — APPROVED** [review]
Thanks @mukeshpanchal27. Let's wait to merge this until after #36 is merged, but this looks good to me.

File: .github/workflows/performance.yml
Comment (adamsilverstein) [inline .github/workflows/performance.yml]: won't this run on every push to the PR? I thought we had decided to run only on merges, has that changed?

File: .github/workflows/performance.yml
Comment (adamsilverstein) [inline .github/workflows/performance.yml:55]: should we add a conditional so this only runs when the PR has been merged, eg. if: `github.event.pull_request.merged == true`

File: .github/workflows/performance.yml
Comment (joemcgill) [reply .github/workflows/performance.yml]: The intent is to run the workflow on every push, but only log performance to the dashboard on a push to trunk, so you're right that this probably needs to be more strict. The Gutenberg repo appears to only be running the workflow during pushes [if the push is to the trunk branch](https://github.com/WordPress/gutenberg/blob/trunk/.github/workflows/performance.yml#L8).

File: .github/workflows/performance.yml
Comment (joemcgill) [reply .github/workflows/performance.yml:55]: I don't think so, I think running the workflow before merges is useful info during a review, but we only want to log the data when the PR is merged.

File: .github/workflows/performance.yml
Comment (felixarntz) [reply .github/workflows/performance.yml]: Good point. I think for now we should ensure this only ever runs for `trunk`, not other point release branches.

But also I don't understand why this line here is being added. We do want to restrict pushing the metrics to Code Health to only the `trunk` branch for `push` events, but what value is there in running the rest of this workflow if that is not happening?

I would think we should rather control the entire workflow to only run on `push` events to `trunk`, not just this single step.

File: .github/workflows/performance.yml
Comment (felixarntz) [inline .github/workflows/performance.yml:16]: See the other review conversation, I think we should remove everything from here for now except lines 4-6 and 12-14, so that this workflow only runs on `push` and `pull_request` events for `trunk`.

Covering point releases is a good future enhancement, but we would need to determine first how we can make that work with the Code Health dashboard, as it would probably need multiple instances for those different branches.

**Review by felixarntz — COMMENTED** [review]
@joemcgill @mukeshpanchal27 Left a bit of feedback here to reduce the workflow to running only on `push` and `pull_request` events for `trunk`.

Let's try to get this merged today alongside #36 so that we can open a first version of the core PR.

File: .github/workflows/performance.yml
Comment (felixarntz) [reply .github/workflows/performance.yml]: Actually, never mind my feedback here, I now realize the `github.event_name = 'push'` here is right since we should also run the workflow on PRs, but not report the data to Code Health.

@joemcgill Do you think there is value of for now commenting out this line and putting a comment above to re-add it prior to merge? This way we can show that the implementation is working via the Code Health dashboard. Otherwise, it would only start sending data once merged I think.

File: .github/workflows/performance.yml
Comment (felixarntz) [reply .github/workflows/performance.yml:55]: @joemcgill See my reply in https://github.com/10up/wordpress-develop/pull/41#discussion_r1119187274: There may be benefits in sending data prior to this PR getting merged, even to show how and _that_ it's working.

File: .github/workflows/performance.yml
Comment (joemcgill) [reply .github/workflows/performance.yml]: > Do you think there is value of for now commenting out this line and putting a comment above to re-add it prior to merge?

I don't think so, the point of this PR is to prepare this PR for merge, so I don't think we should open a PR that isn't mergable. Everything up to this point should be testable in our feature branch prior to merging this PR. We can artificially show that that the last step will send data to the [CoreVitals dasboard](https://www.codevitals.run/) by manually running that step, if we would like, but I don't think we should need to.

File: .github/workflows/performance.yml
Comment (joemcgill) [reply .github/workflows/performance.yml:55]: This is simply following the pattern of the other workflows in the repo, I think it's useful for this to run just like other workflows, but am open to us considering other alternatives. However, I'd love to leave it as is and open this as a conversation on the Core PR so we can get feedback from folks who have been maintaining the other jobs.

File: .github/workflows/performance.yml
Comment (joemcgill) [reply .github/workflows/performance.yml:16]: I agree that we don't want to collect data in the dashboard for now, but I do think it's useful to have these run on feature branches in the mean time so we at least have logs to look back to if a performance improvement/regression gets merged during a backport.

File: .github/workflows/performance.yml
Comment (felixarntz) [reply .github/workflows/performance.yml:55]: @joemcgill My comment here was not about this line of code, rather just addressing what you had said in your comment before mine :)

I think this line of code is good as is for now.

File: .github/workflows/performance.yml
Comment (felixarntz) [reply .github/workflows/performance.yml:16]: @joemcgill I'd argue we should leave that to the discussions on https://core.trac.wordpress.org/ticket/57556 and eventually make a change as part of that ticket.

File: .github/workflows/performance.yml
Comment (felixarntz) [reply .github/workflows/performance.yml]: @joemcgill WFM.

I don't feel strongly about the idea of a "PR that isn't mergable". It will eventually be "manually" committed rather than just merged anyway, and I think calling out things like "Before committing, make minor change x" in bold on the PR description can be reasonable.

If you think our current test dashboard data is enough to show how this is working, let's go with that 👍

File: .github/workflows/performance.yml
Comment (joemcgill) [reply .github/workflows/performance.yml:16]: Sorry, mistakenly said "feature branches" when I meant "release branches". I agree that we should remove the feature branches as we've done and defer conversation about feature branches to that Trac ticket.

**Comment by joemcgill** [comment]
In c7208e0, I've updated the Publish step so it only runs on pushes to `trunk`. Since these PRs will never actually be "merged" with the WordPress PR > Trac commit > sync workflow, I think this should work. Theoretically, forks who push to their `trunk` branch would likely trigger this step as well, but they would need to have the correct project token set as a GitHub secret in order to successfully send data to the CodeVitals dashboard.

I'm going to go ahead and merge this at this point based on @felixarntz's preemptive 👍🏻 on the comment above, so I can check the PR against WordPress's trunk, where we'll be able to review all of this again in it's entirety.

**Comment by felixarntz** [comment]
@joemcgill 
> Theoretically, forks who push to their `trunk` branch would likely trigger this step as well, but they would need to have the correct project token set as a GitHub secret in order to successfully send data to the CodeVitals dashboard.

I think the entire workflow won't run in that case due to the extra condition we have to run only for `WordPress/wordpress-develop` - so I think we're good here either way :)

**Pull request merged**
